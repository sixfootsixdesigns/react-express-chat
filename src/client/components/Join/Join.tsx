import React from 'react';

interface JoinProps {
  addUser: (name: string) => void;
}

interface JoinState {
  username: string;
}

export class Join extends React.Component<JoinProps, JoinState> {
  public state = {
    username: ''
  };

  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      username: event.currentTarget.value
    });
  };

  handleJoin = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    event.preventDefault();
    if (this.state.username) {
      this.props.addUser(this.state.username);
    }
  };

  public render() {
    const { username } = this.state;
    return (
      <div className="JoinWrapper">
        <div className="Join">
          <input
            type="text"
            placeholder="Enter your name"
            required={true}
            onChange={this.handleChange}
            value={username}
          />
          <button onClick={this.handleJoin}>Join</button>
        </div>
      </div>
    );
  }
}
