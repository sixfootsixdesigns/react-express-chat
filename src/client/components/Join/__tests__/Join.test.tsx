import * as React from 'react';
import {
  render,
  fireEvent,
  wait,
  getByText,
  getByPlaceholderText,
  cleanup
} from 'react-testing-library';
import { Join } from '../Join';

describe('Join Component', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render', () => {
    const addUserHandler = jest.fn();

    const { container } = render(<Join addUser={addUserHandler} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not call addUser if no user is entered', async () => {
    const addUserHandler = jest.fn();

    const { container } = render(<Join addUser={addUserHandler} />);

    const button = getByText(container, 'Join');
    fireEvent.click(button);

    await wait(() => {
      expect(addUserHandler).not.toHaveBeenCalled();
    });
  });

  it('should call addUser if user is entered', async () => {
    const addUserHandler = jest.fn();

    const { container } = render(<Join addUser={addUserHandler} />);

    const input = getByPlaceholderText(container, 'Enter your name');
    fireEvent.change(input, {
      target: { value: 'dave' }
    });

    const button = getByText(container, 'Join');
    fireEvent.click(button);

    await wait(() => {
      expect(addUserHandler).toHaveBeenCalledWith('dave');
    });
  });
});
