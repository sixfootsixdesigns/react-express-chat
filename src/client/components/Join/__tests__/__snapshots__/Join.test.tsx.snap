// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join Component should render 1`] = `
<div
  class="JoinWrapper"
>
  <div
    class="Join"
  >
    <input
      placeholder="Enter your name"
      required=""
      type="text"
      value=""
    />
    <button>
      Join
    </button>
  </div>
</div>
`;
